manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -72,41 +76,61 @@ public Bridge createDependencies(Work work, Bridge bridge) {
         rhoasService.createTopicAndGrantAccessFor(resourceNamesProvider.getBridgeErrorTopicName(bridge.getId()),
                 RhoasTopicAccessType.CONSUMER_AND_PRODUCER);
 
-        // We don't need to wait for the Bridge to be READY to create the Error Handler.
-        createErrorHandlerProcessor(bridge);
+        // We don't need to wait for the Bridge to be READY to handle the Error Handler.
+        createOrUpdateOrDeleteErrorHandlerProcessor(bridge);
+
+        // Keep the Bridge in PREPARING until the Error Handler processor is READY.
+        // This ensures Users cannot update the Bridge Error Handler until it is available.
+        if (isErrorHandlerProcessorReady(bridge)) {
```

Comment:
@manstis Checking my understanding here: The `Bridge` will enter the `FAILED` state only after exceeding the time limit for the provisioning process if the ErrorHandler fails to deploy?

If the `ErrorHandler` has failed to deploy, shouldn't we immediately mark the `Bridge` as failed within this loop as well?

cheers,

Rob

Reply:
Well........ you have to go and open a whoop-ar5e can of worms; don't you :laughing: 

The code here would put a `Bridge`'s dependencies into `READY` so that the `Bridge` itself would continue to be re-provisioned by the Operator and the `Bridge` move into `READY` itself. This would happen if the Error Handler was either successfully updated or failed leading to a scenario where the `Bridge` is `READY` but Error Handler `FAILED`. 

Your comment suggests the preferred scenario is that, if the Error Handler fails we short-circuit and fail the `Bridge` too.

I've updated this PR to do the same. It is definitely consistent with parent-child dependencies elsewhere.

Reply:
@manstis That was my understanding of the outcome of the discussions with the other week at the F2F. I thought that we had agreed:

- Failure to deploy the Error Handler would result in the `Bridge` additionally entering the `FAILED` status, as the user has no direct visibility on the Error Handler
- The work you are doing to add a `status_message` to the `Bridge` resource in #968  would provide the user with a bit more information as to why the `Bridge` is in `FAILED` status.

Do you feel anything I'm suggesting here is not in alignment with those discussions?

cheers,

Rob

Reply:
No, no @robpblake I agree and have made the change in this PR.